packages/varlock-website/src/content/docs/integrations/cloudflare.mdx

```diff
@@ -0,0 +1,219 @@
+```
+
+:::note[Future improvement]
+Cloudflare is working on the ability to push secrets with a deploy command. See [this pull request](https://github.com/cloudflare/workers-sdk/pull/10896) for updates.
+:::
+
+---
+
+## Approach 2: Using Wrangler Vars and Secrets (Without Vite)
+
+If you prefer not to use the Vite plugin, you can use Wrangler's built-in environment variable handling. Note that this approach will not provide varlock's security features like leak detection and log redaction.
+
+### Local Development
+
+For local development, you can use `varlock run` to inject resolved environment variables:
+
+```diff lang="json" title="package.json"
+{
+  "scripts": {
+-    "dev": "wrangler dev",
++    "dev": "env -i PATH="$PATH" varlock run -- <command>",
+  }
+}
+```
+
+:::caution
+Because you're creating a child process with a 'clean' environment, you may need to explicitly include any system level env vars that you need.
+:::
+
+### Deployment
+
+For deployments using Wrangler's built-in environment variables, you'll need to use the 3-step deployment process with `secret bulk`:
+
+```bash
+# Step 1: Create a version that's not deployed immediately
+# and note the version id for the next steps
+wrangler versions upload
+
+# Step 2: Attach secrets in bulk (using resolved env in env format)
+wrangler secret bulk --version <version-id> < varlock load --format env
+
+# Step 3: Activate the deployment
+wrangler versions deploy <version-id>
+```
+
+---
+
+## Accessing Environment Variables
+
+Regardless of which approach you use, we recommend using varlock's `ENV` object instead of Cloudflare's built-in environment variable access:
+
+```ts title="src/index.ts"
+import { ENV } from 'varlock/env';
+
+export default {
+  async fetch(request: Request): Promise<Response> {
+    // ❌ Not recommended - uses Cloudflare's built-in env
+    // const apiKey = env.API_KEY;
+
+    // ✅ Recommended - uses varlock's ENV
+    const apiKey = ENV.API_KEY;
+
+    return new Response('Hello World');
+  },
+};
+```
+
+### Why use `ENV` instead of Cloudflare's built-in env?
+
+- Non-string values (e.g., number, boolean) are properly typed and coerced
+- Better error messages for invalid or unavailable keys
+- Type safety and IntelliSense support
+- Security features like leak detection and log redaction (if using the Vite plugin)
+- Consistent API across different deployment platforms
+
+---
+
+## Managing Multiple Environments
+
+Varlock can load multiple _environment-specific_ `.env` files (e.g., `.env.development`, `.env.preview`, `.env.production`) by using the [`@currentEnv` root decorator](/reference/root-decorators/#currentenv).
+
+You can use Cloudflare's branch name to determine the environment:
+
+```env-spec title=".env.schema"
+# @currentEnv=$APP_ENV
+# ---
+WORKERS_CI_BRANCH=
+# @type=enum(development, preview, production, test)
+APP_ENV=remap($WORKERS_CI_BRANCH, production="main", preview=regex(.*), development=undefined)
```

Comment:
I think I want to deprecate remap, or change it anyway... I really dont like the restrictions on keynames.